FV3/atmos_cubed_sphere/tools/coarse_grained_restart_files.F90

```diff
@@ -0,0 +1,366 @@
+
+    filename = 'fv_tracer_coarse.res.nc'
+
+    do n_tracer = 1, n_prognostic_tracers
+       call get_tracer_names(MODEL_ATMOS, n_tracer, tracer_name)
+       call set_tracer_profile(MODEL_ATMOS, n_tracer, restart%q(:,:,:,n_tracer))
+       id_restart = register_restart_field(restart%fv_tracer_coarse, &
+            filename, tracer_name, restart%q(:,:,:,n_tracer), domain=coarse_domain, &
+            mandatory=.false., tile_count=tile_count)
+    enddo
+
+    do n_tracer = n_prognostic_tracers + 1, n_tracers
+       call get_tracer_names(MODEL_ATMOS, n_tracer, tracer_name)
+       call set_tracer_profile(MODEL_ATMOS, n_tracer, restart%qdiag(:,:,:,n_tracer))
+       id_restart = register_restart_field(restart%fv_tracer_coarse, &
+            filename, tracer_name, restart%qdiag(:,:,:,n_tracer), domain=coarse_domain, &
+            mandatory=.false., tile_count=tile_count)
+    enddo
+  end subroutine register_fv_tracer_coarse
+
+  subroutine register_fv_srf_wnd_coarse(tile_count, coarse_domain, restart)
+    integer, intent(in) :: tile_count
+    type(domain2d), intent(in) :: coarse_domain
+    type(coarse_restart_type), intent(inout) :: restart
+
+    character(len=64) :: filename
+    integer :: id_restart
+
+    filename = 'fv_srf_wnd_coarse.res.nc'
+
+    id_restart = register_restart_field(restart%fv_srf_wnd_coarse, &
+         filename, 'u_srf', restart%u_srf, domain=coarse_domain, &
+         tile_count=tile_count)
+    id_restart = register_restart_field(restart%fv_srf_wnd_coarse, &
+         filename, 'v_srf', restart%v_srf, domain=coarse_domain, &
+         tile_count=tile_count)
+  end subroutine register_fv_srf_wnd_coarse
+
+  subroutine register_mg_drag_coarse(tile_count, coarse_domain, restart)
+    integer, intent(in) :: tile_count
+    type(domain2d), intent(in) :: coarse_domain
+    type(coarse_restart_type), intent(out) :: restart
+
+    character(len=64) :: filename
+    integer :: id_restart
+
+    filename = 'mg_drag_coarse.res.nc'
+
+    id_restart = register_restart_field(restart%mg_drag_coarse, &
+         filename, 'ghprime', restart%sgh, domain=coarse_domain, &
+         tile_count=tile_count)
+  end subroutine register_mg_drag_coarse
+
+  subroutine register_fv_land_coarse(tile_count, coarse_domain, restart)
+    integer, intent(in) :: tile_count
+    type(domain2d), intent(in) :: coarse_domain
+    type(coarse_restart_type), intent(inout) :: restart
+
+    character(len=64) :: filename
+    integer :: id_restart
+
+    filename = 'fv_land_coarse.res.nc'
+
+    id_restart = register_restart_field(restart%fv_land_coarse, &
+         filename, 'oro', restart%oro, domain=coarse_domain, &
+         tile_count=tile_count)
+  end subroutine register_fv_land_coarse
+
+  subroutine coarse_grain_restart_data(Atm)
+    type(fv_atmos_type), intent(inout) :: Atm
+
+    if (trim(Atm%coarse_graining%strategy) .eq. MODEL_LEVEL) then
+       call coarse_grain_restart_data_on_model_levels(Atm)
+    endif
```

Comment:
Is there a warning elsewhere if `Atm%coarse_graining%strategy` isn't one of the available options?

Reply:
That's right.  There's a hard error in `coarse_graining_init` if an invalid option is provided:

https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/42f2877a4e04882a1046d35662bce61c85385e27/FV3/atmos_cubed_sphere/tools/coarse_graining.F90#L102-L111

Reply:
Great.